DataDog/datadog-cloudformation-macro
# Add support for Secrets Manager

### What does this PR do?

Add support for AWS Secrets Manager for rotating API keys.

### Motivation

Refer to [SLS-1651](https://datadoghq.atlassian.net/browse/SLS-1651). 

### Testing Guidelines

- Adding test cases to check multiple api keys.
- Not add the key in node runtimes where synchronous metrics is being used, since it's not supported.
- Correctly add the key for every other case.

### Additional Notes

N/A

### Types of changes


### Check all that apply
